pages/tier.tsx

```diff
@@ -0,0 +1,123 @@
+  error: any;
+};
+
+const getPageMetaData = (pageProps: TierPageProps, data) => {
+  let canonicalURL;
+  const tier = data?.Tier;
+  const collective = tier?.collective;
+  const baseMetadata = getCollectivePageMetadata();
+  const { collectiveType, collectiveSlug, parentCollectiveSlug, tierId, tierSlug } = pageProps;
+
+  if (collective) {
+    const collective = tier.collective;
+    canonicalURL = `${getWebsiteUrl()}${getCollectivePageRoute(collective)}/contribute/${tier.slug}-${tier.id}`;
+    return {
+      ...baseMetadata,
+      title: `${collective.name} - ${tier.name}`,
+      description: tier.description || collective.description || collective.longDescription,
+      twitterHandle: collective.twitterHandle || get(collective, 'parentCollective.twitterHandle'),
+      canonicalURL,
+    };
+  } else if ([CollectiveType.EVENT, CollectiveType.PROJECT].includes(collectiveType)) {
+    canonicalURL = `${getWebsiteUrl()}/${parentCollectiveSlug}/${collectiveType}/${collectiveSlug}/contribute/${tierSlug}-${tierId}`;
+  } else {
+    canonicalURL = `${getWebsiteUrl()}/${collectiveSlug}/contribute/${tierSlug}-${tierId}`;
+  }
+
+  return { ...baseMetadata, title: 'Tier', canonicalURL };
+};
+
+/**
+ * The main page to display collectives. Wrap route parameters and GraphQL query
+ * to render `components/collective-page` with everything needed.
+ */
+const TierPage = pageProps => {
+  const router = useRouter();
+  const { LoggedInUser } = useLoggedInUser();
+  const { tierId, tierSlug, data, redirect, error, loading } = pageProps;
+  const collective = data?.Tier?.collective;
+
+  React.useEffect(() => {
+    if (collective) {
+      addParentToURLIfMissing(router, collective, `/contribute/${tierSlug}-${tierId}`);
+    }
+  }, [collective]);
+
+  return !data || error ? (
+    <ErrorPage data={error || data} />
+  ) : (
+    <Page {...getPageMetaData(pageProps, data)}>
+      {loading || !data.Tier || !data.Tier.collective ? (
+        <div className="py-16 sm:py-32">
+          <Loading />
+        </div>
+      ) : (
+        <CollectiveThemeProvider collective={data.Tier.collective}>
+          <TierPageContent
+            LoggedInUser={LoggedInUser}
+            collective={data.Tier.collective}
+            tier={data.Tier}
+            contributors={data.Tier.contributors}
+            contributorsStats={data.Tier.stats.contributors}
+            redirect={redirect}
+          />
+        </CollectiveThemeProvider>
+      )}
+    </Page>
+  );
+};
+
+export const getServerSideProps = async ({
+  query: { parentCollectiveSlug, collectiveSlug, tierId, tierSlug, redirect, collectiveType },
+}): Promise<{ props: TierPageProps }> => {
+  const client = initClient();
+  const { data, error, loading } = await client.query({
+    query: tierPageQuery,
+    variables: { tierId: Number(tierId) },
+  });
+
+  return {
+    props: {
+      // Required
+      collectiveSlug,
+      tierId: Number(tierId),
+      tierSlug,
+      loading,
```

Comment:
I think we don't need to pass "loading" here. Since getServerSideProps is only called in the backend, we'll always have either data or error populated and loading should probably always be false.
```suggestion
```